astro/src/content/blog/forgerock-password-stroge.mdx

```diff
@@ -0,0 +1,114 @@
+
+This is a very simplified example. The next letter hashing algorithm is not sophisticated and has many issues such as you can see the passwords would start with the same letter and you can find the hashed letters for 'cat' in both. The hashing functions mentioned above solve these problems and many others making the one way hash secure.
+
+### Iterations
+
+An iteration is just a repetition of the process. In our example above, one iteration of our hashing function with the password '**cat**' produced '*dbu*'. If you were to perform a second iteration, the hash would become '*ecv*', ('**d**' becomes '*e*', '**b**' becomes '*c*', '**u**' becomes '*v*'). You can do this as many times as necessary to fit your requirements. It is important to note the number of times this happens with a hash so that you can be sure to compare the correct iteration of the password hash to the one stored.
+
+## How Do I Use It To Migrate Users
+
+Secure authorization systems will not store user passwords. They will only store the hash. If they used a salted hash, they will likely store the salt as well. By moving the hash, the salt and the number of iterations to another system, you should be able to derive the same hash with the given salt if you are using the same hashing algorithm. When a user enters their password on your site, apply the salt and apply the same hashing algorithm the same number of times that was used in the old system. Then compare the hash you get with the stored hash. They will match if the same password was used. This is how you are able to migrate a user without ever needing to know the actual password. 
+
+## How Do I Decipher Forgerock's Hash
+
+Password hash and salt storage will vary between implementations. Hashing algorithms will vary as well and could even vary between users. By default, Forgerock uses the PBKDF2-HMAC-SHA256 encryption with their Cloud Development Kit so that will be used in this article.
+
+<Aside type="note">
+If your implementation is using a different hashing function, Forgerock has more details on their other password storage schemes [here](https://backstage.forgerock.com/knowledge/kb/article/a44757687).
+</Aside>
+
+Forgerock stores that hash in the following way: 
+
+"&#123;PBKDF2-HMAC-SHA256&#125;"&lt;iterations&gt;":" base64(&lt;digest&gt; &lt;salt&gt;) where:
+
+- &#123;PBKDF2-HMAC-SHA256&#125; is a literal describing the hashing function used.
+- &lt;iterations&gt; is the number of times the algorithm is applied
+- ":" is a literal separator
+- base64(&lt;digest&gt; &lt;salt&gt;) indicates that the password hash and salt are stored together and base64 encoded
+
+In the example:
+
+&#123;PBKDF2-HMAC-SHA256&#125;10:8c7nLGEIXeZf45YQ92A2MD+v8olvKKl6iWXGQZoluJ/awqZnHwFvslIOx7xOZ9AV
+  
+  - PBKDF2-HMAC-SHA256 is the hashing function used
+  - 10 is the number of iterations
+  - 8c7nLGEIXeZf45YQ92A2MD+v8olvKKl6iWXGQZoluJ/awqZnHwFvslIOx7xOZ9AV is the Base64 encoded password hash and salt
```

Comment:
```suggestion
  - `8c7nLGEIXeZf45YQ92A2MD+v8olvKKl6iWXGQZoluJ/awqZnHwFvslIOx7xOZ9AV` is the Base64 encoded password hash and salt
```

Reply:
I don't know if we have a standard capitalization/casing for base 64. Please find one and make it consistent in this post and add it to DevDocsREADME.

Reply:
Base64 endoding/decoding is case specific. We should leave as is and will update the DevDocsREADME with that info.

Reply:
In the RFC it uses Base64 in the title but the test it states: `This encoding may be referred to as "base64"`. There for it is lowercase unless in a title.